Sources/MockoloFramework/Parsers/SwiftSyntaxExtensions.swift

```diff
@@ -738,42 +751,77 @@ final class EntityVisitor: SyntaxVisitor {
     }
 
     override func visit(_ node: ImportDeclSyntax) -> SyntaxVisitorContinueKind {
-        if let ret = node.path.firstToken(viewMode: .sourceAccurate)?.text {
-            let desc = node.importKeyword.text + " " + ret
-            imports["", default: []].append(desc)
-        }
+        imports.append(
+            .init(line: node.trimmedDescription)
+        )
         return .skipChildren
     }
 
     override func visit(_ node: IfConfigDeclSyntax) -> SyntaxVisitorContinueKind {
-        for cl in node.clauses {
-            let macroName: String
+        let blockId = Self.ifConfigDeclCount
+        Self.ifConfigDeclCount += 1
+        var previousConditions: [String] = []
+
+        for (index, cl) in node.clauses.enumerated() {
+            let compilerDirectiveKey: String
+
             if let conditionDescription = cl.condition?.trimmedDescription {
-                macroName = conditionDescription
+                if index == 0 {
+                    guard conditionDescription != fileMacro else {
+                        return .visitChildren
+                    }
+                    compilerDirectiveKey = "\(conditionDescription):\(blockId):if"
+                } else {
+                    compilerDirectiveKey = "\(conditionDescription):\(blockId):elseif-\(index)"
+                }
+                previousConditions.append(conditionDescription)
             } else {
-                return .visitChildren
+                if let previousCondition = previousConditions.first {
+                    compilerDirectiveKey = "\(previousCondition):\(blockId):else"
+                } else {
+                    return .visitChildren
+                }
             }
 
-            guard macroName != fileMacro else { return .visitChildren }
-
             if let list = cl.elements?.as(CodeBlockItemListSyntax.self) {
                 for el in list {
+                    let nestedImport: ImportStatement
                     if let importItem = el.item.as(ImportDeclSyntax.self) {
-                        let key = macroName
-                        if imports[key] == nil {
-                            imports[key] = []
-                        }
-                        imports[key]?.append(importItem.trimmedDescription)
-
+                        nestedImport = .init(
+                            line: importItem.trimmedDescription,
+                            compilerDirectiveKey: compilerDirectiveKey
+                        )
                     } else if let nested = el.item.as(IfConfigDeclSyntax.self) {
```

Comment:
With this approach, you can only handle nested `#if` one time.  
I think it should be a recursive structure to work correctly.  
However, since this spec is from the beginning, maybe you can ignore it for now.

Input:

```swift
#if canImport(NewFramework)
import Z
#if canImport(NewFramework2)
import W
#if canImport(NewFramework3)
import Z
#endif
#endif
#endif
```

Current output(`import Z` has gone) :

```swift
#if canImport(NewFramework)
#if canImport(NewFramework2)
import W
#endif
import Z
#endif
```